async-nats/src/lib.rs

```diff
@@ -1293,7 +1293,7 @@ impl std::fmt::Display for ServerError {
 }
 
 /// Info to construct a CONNECT message.
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Deserialize, Serialize)]
```

Comment:
I don't think there is any benefit in addint any addional traits to ConnectInfo.
This value is only passed to the server. To be fair, it could be private.

Reply:
I don't need this quite yet, but was intending on caching them on the server per-connection, and will be deserializing them on the server.